cmake/create_test.cmake

```diff
@@ -100,7 +164,7 @@ function(ginkgo_create_dpcpp_test test_name)
     target_compile_options(${test_target_name} PRIVATE ${GINKGO_DPCPP_FLAGS})
     target_link_options(${test_target_name} PRIVATE -fsycl-device-code-split=per_kernel)
     ginkgo_set_test_target_properties(${test_target_name} ${ARGN})
-    ginkgo_add_test(${test_name} ${test_target_name} ${ARGN})
+    ginkgo_add_test(${test_name} ${test_target_name} ${ARGN} RESOURCE_TYPE syclgpu)
```

Comment:
we have syclcpu testing

Reply:
For that, we just ignore the resource config

Reply:
Since we should be able to run CPU tests anywhere, I think it would be best if we allowed setting the resource type to CPU for SYCL, ideally setting it to the number of cores that SYCL would be using (I'm sure we'll figure out how to determine that)

Reply:
I think the first question is if we can determine during cmake, if it will run sycl on cpu or gpus.
Regarding setting the number of cores, maybe `ONEAPI_DEVICE_SELECTOR` (https://github.com/intel/llvm/blob/sycl/sycl/doc/EnvironmentVariables.md#oneapi-device-selector) is the one we want. But I don't know how well that is supported.

Reply:
I guess, we need to wrap our dpcpp test in a script to set this variable from the ctest variables.

Reply:
I would use either a variable `GINKGO_DPCPP_TEST_GPU_RESOURCES` or indeed parse the SYCL device list to check whether there is a GPU available.

Reply:
Now the resource is just called `sycl` and represents either a SYCL GPU or CPU, that should resolve this question.